context/src/main/java/io/opentelemetry/context/ThreadLocalContextStorage.java

```diff
@@ -45,7 +41,8 @@ public Scope attach(Context toAttach) {
 
   @Override
   public Context current() {
-    return THREAD_LOCAL_STORAGE.get();
+    Context threadContext = THREAD_LOCAL_STORAGE.get();
+    return threadContext != null ? threadContext : Context.root();
```

Comment:
Changes in this file seem independent of the main issue, consider to split them.

Reply:
Done and done, #1974